cluster-endpoints/src/multiplexing_subscriptions.rs

```diff
@@ -0,0 +1,112 @@
+use crate::endpoint_stremers::EndpointStreaming;
+use anyhow::Context;
+use prometheus::{opts, register_gauge, Gauge};
+use solana_lite_rpc_core::{commitment_utils::Commitment, AnyhowJoinHandle};
+use solana_sdk::slot_history::Slot;
+use std::collections::BTreeSet;
+
+lazy_static::lazy_static! {
+    static ref GRPC_SLOT_UPDATE: Gauge =
+    register_gauge!(opts!("literpc_rpc_slot_update_from_grpc", "Is slot updated by grpc notification")).unwrap();
+
+    static ref GRPC_BLOCK_UPDATE: Gauge =
+    register_gauge!(opts!("literpc_rpc_block_update_from_grpc", "Is block updated by grpc notification")).unwrap();
+}
+
+const NB_BLOCKS_TO_CACHE: usize = 1024;
+
+pub fn multiplexing_endstreams(
+    rpc_endpoints: EndpointStreaming,
+    grpc_endpoint: EndpointStreaming,
+) -> anyhow::Result<(EndpointStreaming, Vec<AnyhowJoinHandle>)> {
+    let (slot_sx, slot_notifier) = tokio::sync::broadcast::channel(10);
+    let (block_sx, blocks_notifier) = tokio::sync::broadcast::channel(10);
+    let mut endpoint_tasks = vec![];
+
+    let mut rpc_slot_notifier = rpc_endpoints.slot_notifier;
+    let mut grpc_slot_notifier = grpc_endpoint.slot_notifier;
+    let slot_multiplexer: AnyhowJoinHandle = tokio::spawn(async move {
+        let mut processed_slot = 0;
+        let mut estimated_slot = 0;
+        loop {
+            let (notification, is_grpc_update) = tokio::select! {
```

Comment:
pls to not use bool for `is_grpc_update` but enum instead 
(https://corrode.dev/blog/enums/)

Reply:
This makes sense if the boolean is in struct, but does not if it is a variable which is later used.